### ISSUE TITLE
Add docs for how to define a new Concrete Term type

### ISSUE THREAD

**Issue by oxinabox** [JuliaStats/StatsModels.jl#121]
I am fairly sure this is easy to do,
though it will maybe never be the default term,
You will be able to specify it via hints?

Might be good to do an example of `AngualarTerm` or `CyclicalContinousTerm` that represents an angle and is modelcol'd to `sin(X),cos(x)`

Bonus points if it can actually be made the default term type for Unitful angles

**Comment by kleinschmidt** [comment]
yeah it's simple, and you can do it with only "light" type piracy by creating method for `concrete_term` that return teh correct concrete term type, like 

```julia
concrete_term(t::Term, xs::Vector{MyUnitfulType}, ::Nothing) = MyUnitfulTerm(t.sym, my_summary_function(xs))
```

(`nothing` is what's provided as a "hint" when it's not manually set: https://github.com/JuliaStats/StatsModels.jl/blob/master/src/schema.jl#L167)
